File: app/src/main/java/com/example/shenjack/zhihudailyreader/topstories/TopFragmentPagerAdapter.java
package com.example.shenjack.zhihudailyreader.topstories;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.shenjack.zhihudailyreader.R;
import com.example.shenjack.zhihudailyreader.data.TodayPosts;
import com.example.shenjack.zhihudailyreader.stoylist.StoryListAdapterOnClickListener;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

import butterknife.ButterKnife;

import static android.content.ContentValues.TAG;

/**
 * Created by ShenJack on 2017/6/21.
 */

public class TopFragmentPagerAdapter extends FragmentPagerAdapter {
    private List<TodayPosts.TopStoriesBean> topStories;
    Context mContext;
    private StoryListAdapterOnClickListener storyListAdapterOnClickListener;
    private List<View> views;

    public TopFragmentPagerAdapter(FragmentManager fm
            , Context context
            , List<TodayPosts.TopStoriesBean> topStories
            , StoryListAdapterOnClickListener storyListAdapterOnClickListener) {
        super(fm);
        mContext = context;
        this.topStories = topStories;
        this.storyListAdapterOnClickListener = storyListAdapterOnClickListener;
        views = new ArrayList<>();
        initPages();
    }

    @Override
    public Fragment getItem(int position) {
        return NewFragment.newInstance(views.get(position));
    }

    @Override
    public int getCount() {
        return topStories.size();
    }

    public void setTopStories(List<TodayPosts.TopStoriesBean> topStories) {
        this.topStories = topStories;
        initPages();
        notifyDataSetChanged();
    }

    private void initPages(){
        for (TodayPosts.TopStoriesBean topStoriesBean :
                topStories) {
            View view = LayoutInflater.from(mContext).inflate(R.layout.top_thumbnail, null);
            ImageView headerImage = ButterKnife.findById(view, R.id.thumbnail_image);
            final String title = topStoriesBean.getTitle();
            final TextView titleTextView = ButterKnife.findById(view, R.id.title);
            final int storyId = topStoriesBean.getId();
            URL url = null;
            try {
                url = new URL(topStoriesBean.getImage());
            } catch (MalformedURLException e) {
                e.printStackTrace();
            }
            titleTextView.setText(title);

            Glide.with(mContext).load(url).into(headerImage);

            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    storyListAdapterOnClickListener.onClick(storyId, title);
                }
            });

            Log.d(TAG, "initPages: title:" + title);

            views.add(view);
        }
    }
}

File: app/src/main/java/com/example/shenjack/zhihudailyreader/topstories/TopFragmentPagerAdapter.kt
package com.example.shenjack.zhihudailyreader.topstories

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.bumptech.glide.Glide
import com.example.shenjack.zhihudailyreader.R
import com.example.shenjack.zhihudailyreader.data.TodayStories
import com.example.shenjack.zhihudailyreader.stoylist.StoryListAdapterOnClickListener

import java.net.MalformedURLException
import java.net.URL
import java.util.ArrayList

import butterknife.ButterKnife


/**
 * Created by ShenJack on 2017/6/21.
 */

class TopFragmentPagerAdapter(fm: FragmentManager, internal var mContext: Context, private var topStories: List<TodayStories.TopStoriesBean>?, private val storyListAdapterOnClickListener: StoryListAdapterOnClickListener) : FragmentPagerAdapter(fm) {
    private val views: MutableList<View>

    init {
        views = ArrayList<View>()
        initPages()
    }

    override fun getItem(position: Int): Fragment {
        return TopFragment.newInstance(views[position])

    }

    override fun getCount(): Int {
        return topStories!!.size
    }

    fun setTopStories(topStories: List<TodayStories.TopStoriesBean>) {
        this.topStories = topStories
        initPages()
        notifyDataSetChanged()
    }

    private fun initPages() {
        for (topStoriesBean in topStories!!) {
            val view = LayoutInflater.from(mContext).inflate(R.layout.top_thumbnail, null)
            val headerImage = ButterKnife.findById<ImageView>(view, R.id.thumbnail_image)
            val title = topStoriesBean.title
            val titleTextView = ButterKnife.findById<TextView>(view, R.id.title)
            val storyId = topStoriesBean.id
            var url: URL? = null
            try {
                url = URL(topStoriesBean.image)
            } catch (e: MalformedURLException) {
                e.printStackTrace()
            }

            titleTextView.text = title

            Glide.with(mContext).load(url).into(headerImage)

            view.setOnClickListener { storyListAdapterOnClickListener.onClick(storyId, title!!) }


            views.add(view)
        }
    }
}
